### ISSUE TITLE
Database - Loading large number of posts/pagination

### ISSUE THREAD

**Issue by CreaRo** [firebase/quickstart-android#160]
Thanks for this repo, its extremely helpful. I'm sorry my issue is very verbose, but I simply can't find a usable solution anywhere.

tldr; Pagination. I need help. Please don't redirect me to stackoverflow, I've Googled as much as I could.

I'm following the database code and have setup my adapters, viewholder and everything works perfectly.
I'm using firebase for the first time on an app that has >50K active users and would really like to perfect my usage prior to releasing it on the playstore.

However, now I realize how conveniently this example has ignored large snapshot sizes. The 'Recent' tab limits its results to the top 100 only. 'My posts' and 'my Top Posts' won't ever exceed a hundred. But what about having an 'All posts' tab? 

So here's exactly where my problem is. My app allows users to create quotes (motivational quotes etc), and displays them in these tabs. Now with 55K active users and growing, I can expect at least 100 new posts a day. Clearly, I need some sort of **pagination** to allow me to have an 'All posts' tab, and I can't find an example anywhere. I've gone through each stackoverflow question, and none of them really help - how would I implement them in a `FirebaseRecyclerAdapter`? I can't change the `Query ref` in the constructor of the adapter, there isn't a method to modify that anywhere - so `.startsAt()` and `.limitFirst()` etc etc methods can't be used there. 

Any example/code snippet where the FirebaseRecyclerAdapter is used and has implemented working pagination?

**Comment by CreaRo** [comment]
Just found [this](https://github.com/firebase/FirebaseUI-Android/issues/110) issue. I think I'll be able to use the open [PR](https://github.com/firebase/FirebaseUI-Android/pull/26). I'll also hopefully write up a short blog soon, on using pagination and post here for anyone to use!
It's strange there still isn't a clear paginating mechanism in Firebase - I've seen posts from 2013 where the team has been saying its on the todo list. It really is high time Firebase gets a layman solution for pagination!
